tests/ThreeWayMergeTest.php

```diff
@@ -196,12 +196,648 @@ public function testConflict()
+    }
+
+    public function testMultilineAdditionRemovalModificationTest6()
+    {
+        $original = [
+            'keyA' => ['This is not fun
+            I like doing it
+            it was easy'],
+        ];
+
+        $local = [
+            'keyA' => ['This is fun'],
+        ];
+
+        $remote = [
+            'keyA' => ['This is not fun
+            I like doing it'],
+        ];
+
+        $expected = [
+            'keyA' => ['This is fun
+            it was easy'],
+        ];
+        $merge = new ThreeWayMerge();
+        $result = $merge->performMerge($original, $local, $remote);
+        $this->assertEquals($expected, $result);
+    }
+    public function testMultilineRecursiveAdditionRemovalModification()
+    {
+        $original = [
+            'keyA' => [
+                0 => [
+                    'keyB' => 'This is honey
+                    and who cares',
+                    'keyC' => 'This is however, not apple',
+                ],
+                1 => [
+                    'keyB' => 'This is milk',
+                    'keyC' => 'This is mango',
+                ],
+                2 => 'a little sugar
+                this line will be removed and you wont see it',
+            ]
+        ];
+        $local = [
+            'keyA' => [
+                0 => [
+                    'keyB' =>  'This is honey',
+                    'keyC' => 'This is however, not apple',
+                ],
+                2 => 'a little sugar',
+                1 => [
+                    'keyB' => 'This is milky milky',
+                    'keyC' => 'This is mango',
+                ],
+            ]
+        ];
+        $remote = [
+            'keyA' => [
+                0 => [
+                    'keyB' => 'This is updated and purified honey
+                    and who cares
+                    i dont',
+                    'keyC' => 'This is apple',
+                ],
+                1 => [
+                    'keyB' => 'This is milk
+                    and I like milk',
+                    'keyC' => 'This is changed because of remote',
+                ],
+                2 => 'a little new something',
+            ]
+        ];
+
+        $expected = [
+            'keyA' => [
+                0 => [
+                    'keyB' => 'This is updated and purified honey
+                    i dont',
+                    'keyC' => 'This is apple',
+                ],
+                1 => [
+                    'keyB' => 'This is milky milky
+                    and I like milk',
+                    'keyC' => 'This is changed because of remote',
+                ],
+                2 => 'a little new something',
+            ]
+        ];
+        $merge = new ThreeWayMerge();
+        $result = $merge->performMerge($original, $local, $remote);
+        $this->assertEquals($expected, $result);
+    }
+    // Conflict in KeyA=>0=>KeyB.
+    public function testMultilineConflict1()
```

Comment:
Let's add proper documentation block to each test method, briefly describing the test. For example like this:

```php
/**
 * Test that assert conflicts when a key are modified in both versions.
 */
```